### PR TITLE
add travis-ci.org and conda package functionality

### PR DESCRIPTION
**Pull request by avinson**

### CODE REVIEW

**Comment by fsaad** [comment]
@avinson Thanks for the pull request on this and other probcomp repos. The conda build should be helpful for distribution. Here is a question:

How feasible would it be for these `meta.yaml` files:

https://github.com/probcomp/bayeslite/blob/20171023-avinson-travis/conda/meta.yaml
https://github.com/probcomp/cgpm/blob/20171023-avinson-travis/conda/meta.yaml
https://github.com/probcomp/crosscat/blob/20171023-avinson-travis/conda/meta.yaml
https://github.com/probcomp/Venturecxx/blob/20171023-avinson-travis/conda/meta.yaml

to agree exactly on the versions of all overlapping packages (such as matplotlib, numpy, scipy, and the sort?) My sense is that it should be straightforward because the meta.yaml file for the "probcomp/notebook" repo is going to build this whole shebang together in a single conda environment (although let me know if I'm mistaken regarding feasibility of version agreement).

**Comment by avinson** [comment]
@fsaad please review again. versions should match now.. also, i've updated the Dockerfile to reflect the dependencies required by probcomp: https://github.com/probcomp/notebook/blob/master/Dockerfile

**Comment by fsaad** [comment]
@avinson Thank your for unifying the versions. How did you determine the python packages versions to pin to? It seems the python packages in meta.yaml files and Dockerfile at https://github.com/probcomp/notebook/blob/master/Dockerfile downgrade our current target versions (except for cython). In particular, our development environments, and probcomp-4 test server (which includes integration tests and notebook teaching material) are written to target:

numpy==1.11.0
matplotlib==1.5.1
cython==0.23.4
scipy==0.17.0
pandas==0.17.1
scikit-learn==0.17
seaborn==0.7.1
statsmodels==0.6.1
pytest==2.8.7

Could you please confirm whether (modulo point releases) your new build process is compatible with these versions, and update the conda environment to match them? For each package version in the list above that we cannot use in the conda environment, please let me know why and we can work on a solution.

**Comment by avinson** [comment]
discussed offline but will test PRs against versions mentioned above

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+language: python
+dist: trusty
```
Comment (fsaad) [inline .travis.yml]: Please use xenial (16.04) instead of trusty.

Comment (avinson) [reply .travis.yml]: @fsaad there is currently no xenial build environment on travis: https://docs.travis-ci.com/user/reference/overview/

will get back to you on the updated versions for meta.yaml shortly

**Comment by avinson** [comment]
fyi, numpy 0.11 is not satisfiable with pandas 0.17

```
INFO:conda_build.metadata:Attempting to finalize metadata for cgpm
failed to get install actions, retrying.  exception was: The following specifications were found to be in conflict:
  - numpy 1.11.*
  - pandas 0.17.* -> numpy 1.10*
```

**Comment by avinson** [comment]
@fsaad please review again.. with pandas 0.18 and point releases for some of the other deps, package is now buildable and tests pass

**Pull request merged**